### ISSUE TITLE
Invitations - Improve sorting options

### ISSUE THREAD

**Issue by Jackseed** [blockframes/blockframes#4179]
Currently it's a mess for sales agents to filter the invitations they have when organising multiple screenings. We should think of a better organisation so they can sort by events and invitation status

**Comment by jdrns** [comment]
[Work in progress here](https://www.figma.com/file/yJzsxacuRCznxJtC7QYBUc/Invitation-Sorting-Impovment?node-id=0%3A1)

**Review needed:** 
- Business review

**Comment by Jackseed** [comment]
I'll need a dev review to see how it's possible to make these sorting options. On a usability point of view it does look convenient but I'm a bit worried that we may have to create a lot of custom logic to make this possible. 

What about changing the list component to a table? Then all sorting options are built in.

Another idea would be to cut the list in different sections by default? either with tabs as in the dashboard movie list or within the page (same as this material example:
![image](https://user-images.githubusercontent.com/14964443/114392868-a6763b80-9b99-11eb-99a9-226382293c3b.png)
)

Though I'm not sure it will save us the custom logic and thus your proposition would be better

**Comment by maltmayer** [comment]
V1 from the Figma wireframes looks like the best option but it does look quite complicated technically (as results should keep a logic even with the applied" filters) so for example, even if we sort by type or event name, the result we get should still appear in chronological order. And if we choose to filter by event name, then the event names should be sorted alphabetically for example.

V2 (with the tabs) looks okay too but then, there should be a way to show that there are pending invitation (like a red bubble or something) otherwise I fear that the tabs that are not showing up first will never be consulted.

I'm not a fan of the table option

**Comment by Jackseed** [comment]
@GrandSchtroumpf could you have a look to this issue and give us your point of view

**Comment by GrandSchtroumpf** [comment]
We need to differentiate: 
- **Filtering**: remove elements that don't match filters
- **Sorting**: order elements with one criteria

We have to be careful with **filtering** because of deep calls (one invitation item calls the collection "users", "organizations", "events"). Therefore, maybe it's not possible to filter the list based on those collections.

Concerning **sorting**, the current design is very technical and quite messy. For me, there is no way we could develop and maintain one design per sorting method.

Table seems to be easier on the sorting level, but then we enter the loading issue we discussed with @2amarkina last time: a call to another collection (organization, event, user) should be isolated to its own column. Also it's sure we won't be able to filter based on other collection with table.

To sum up: 
- List: filtering seems ok, but sorting have to stay the same.
- Table: sorting seems ok, but filtering is not possible.

**Comment by jdrns** [comment]
[Here](https://www.figma.com/file/yJzsxacuRCznxJtC7QYBUc/Invitation-Sorting-Impovment?node-id=0%3A1) is the new proposition with the list format and two filters: 
- By Invitation type (Events / Organization) 
- By Status (Pending / Accepted / Declined) 

As well as: 
- The search bar discussed during the last sprint retro

**Comment by Jackseed** [comment]
Looks good to me 👍 
@GrandSchtroumpf mb?

**Comment by maltmayer** [comment]
No comment, this looks good :)

**Comment by jdrns** [comment]
Mock up for [invitation page](https://zpl.io/VxpeXyk) (and 'sort by' options)

**Comment by Jackseed** [comment]
@RemcoSimonides we need a new issue for the 3 lines invitations (@jdrns already have some propositions)

**Comment by RemcoSimonides** [comment]
Dev issues created ☝️
